sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayServiceConfigurationReader.java

```diff
@@ -41,118 +41,32 @@
  */
 public class GatewayServiceConfigurationReader {
 
-    public static final String GATEWAY_READER_NOT_INITIALIZED = "GatewayServiceConfigurationReader has not been initialized";
-
-    public ReplicationPolicy userReplicationPolicy;
-    private ReplicationPolicy systemReplicationPolicy;
-    private ConsistencyLevel consistencyLevel;
-    private volatile boolean initialized;
     private URI serviceEndpoint;
-    private final ConnectionPolicy connectionPolicy;
-    private Map<String, Object> queryEngineConfiguration;
-    private final BaseAuthorizationTokenProvider baseAuthorizationTokenProvider;
-    private final boolean hasAuthKeyResourceToken;
-    private final String authKeyResourceToken;
-    private HttpClient httpClient;
+    private GlobalEndpointManager globalEndpointManager;
 
-    public GatewayServiceConfigurationReader(URI serviceEndpoint, boolean hasResourceToken, String resourceToken,
-            ConnectionPolicy connectionPolicy, BaseAuthorizationTokenProvider baseAuthorizationTokenProvider,
-            HttpClient httpClient) {
+    public GatewayServiceConfigurationReader(URI serviceEndpoint, GlobalEndpointManager globalEndpointManager) {
         this.serviceEndpoint = serviceEndpoint;
-        this.baseAuthorizationTokenProvider = baseAuthorizationTokenProvider;
-        this.hasAuthKeyResourceToken = hasResourceToken;
-        this.authKeyResourceToken = resourceToken;
-        this.connectionPolicy = connectionPolicy;
-        this.httpClient = httpClient;
+        this.globalEndpointManager = globalEndpointManager;
+        this.globalEndpointManager.getDatabaseAccountFromCache(this.serviceEndpoint).block();
     }
 
     public ReplicationPolicy getUserReplicationPolicy() {
-        this.throwIfNotInitialized();
-        return this.userReplicationPolicy;
+        DatabaseAccount databaseAccount = this.globalEndpointManager.getDatabaseAccountFromCache(this.serviceEndpoint).block();
+        return BridgeInternal.getReplicationPolicy(databaseAccount);
     }
 
     public ReplicationPolicy getSystemReplicationPolicy() {
-        this.throwIfNotInitialized();
-        return this.systemReplicationPolicy;
-    }
-
-    public boolean enableAuthorization() {
-        return true;
+        DatabaseAccount databaseAccount = this.globalEndpointManager.getDatabaseAccountFromCache(this.serviceEndpoint).block();
+        return BridgeInternal.getSystemReplicationPolicy(databaseAccount);
     }
 
     public ConsistencyLevel getDefaultConsistencyLevel() {
-        this.throwIfNotInitialized();
-        return this.consistencyLevel;
-    }
-
-    public void setDefaultConsistencyLevel(ConsistencyLevel value) {
-        this.throwIfNotInitialized();
-        this.consistencyLevel = value;
+        DatabaseAccount databaseAccount = this.globalEndpointManager.getDatabaseAccountFromCache(this.serviceEndpoint).block();
```

Comment:
blocking calls inside the SDK are not good. they can cause deadlock, perf hit, etc. We are not doing blocking calls anywhere else.

Reply:
done